src/main/java/at/ac/tuwien/kr/alpha/common/depgraph/DependencyGraphUtils.java

```diff
@@ -0,0 +1,147 @@
+package at.ac.tuwien.kr.alpha.common.depgraph;
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+// TODO internalize into DependencyGraph - most stuff here depends on the dependency graph
```

Comment:
Please address this todo: either move into `DependencyGraph` or keep it outside but give it a meaningful name, e.g. `DependencyGraphAlgorithms`